### PR TITLE
Break out the generated postgresql bindings into a seperate crate

### PR DESCRIPTION
**Pull request by posix4e**
Hilariously, since cargo is fundamentally a source distribution mechanism, we can abstract the decision of what generated library is used, to build.rs time.

By default lib.rs won't compile until the the version of posgresql is chosen. This choice will not be automatically checked into the git repo via the gitignore feature. The reviewer should consider if this is enough protection? Should we default to pg95.rs or is that less safe?

### CODE REVIEW

File: rpgffi/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[package]
+name = "rpgffi"
+version = "0.0.7"
```
Comment (solidsnack) [inline rpgffi/Cargo.toml]: Maybe this should be 0.0.1?

File: util/choose_postgresql_version [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -ex
+
+pg_short_version=pg"`pg_config --version | cut -d' ' -f2 | tr -d . | cut -c-2`"
+unlink src/lib.rs || true 
+(cd src; ln -s $pg_short_version.rs lib.rs)
```
Comment (solidsnack) [inline util/choose_postgresql_version]: This appeals to my sense of humour.

Comment (posix4e) [inline util/choose_postgresql_version]: That's how i felt about it

**Comment by posix4e** [comment]
I just figured out how to do this in a cleaner way

File: util/choose_postgresql_version (hunk 2 above)
Comment (posix4e) [inline util/choose_postgresql_version]: An alternative to this commit is an enhancement over what we discussed over the phone. If we had a wrapping rpg library, we could do this simply by setting the feature. See the autogenerated branch for an example.

You can build a pg94 version with
cargo build --features rpg/rpg94. I was mistaken before that this was a problem. The issue there is now we wouldn't get this cleaner calling convention for calling the raw library. In addition, we would have to go back to #cfg trickery if we wanted to call the native libraries (as opposed to the cleaner code in this commit)

**Comment by posix4e** [comment]
Probably the best approach til/if we have a compelling rpg library is to stick to this

File: .gitignore [hunk 3]
```diff
@@ -9,6 +9,11 @@
 
 # Generated by Cargo
 /target/
+rpgffi/target/
 
 # our outgenerated bindings
-src/libpq.rs
+pgffi/src/lib.rs
```
Comment (solidsnack) [inline .gitignore]: `rpgffi/src/lib.rs`?

**Comment by solidsnack** [comment]
I am going to try this out on Monday.

File: .gitignore (hunk 3 above)
Comment (posix4e) [inline .gitignore]: weird typo. You are correct

**Pull request merged**